### PR TITLE
vscode: New window, command palette, editor split

### PR DESCRIPTION
**Pull request by auphofBSF**
Further voice commands, for vscode under windows

### CODE REVIEW

File: code/ide.py [hunk 1]
```diff
@@ -266,6 +275,9 @@ def ide_git_annotate():
     def ide_git_menu():
         """Show the git menu for IDE"""
 
+    def ide_toggle_explorer():
```
Comment (knausj85) [inline code/ide.py:278]: Any reason not to just use ide_toggle_project?

File: code/ide.py [hunk 2]
```diff
@@ -499,3 +511,6 @@ def ide_show_debug():
 
     def ide_show_extensions():
         """Show Extensions"""
+
+    def ide_window_new():
```
Comment (knausj85) [inline code/ide.py:515]: I don't think this new action is necessary, should be able to override the talon-declared one

https://github.com/knausj85/knausj_talon/blob/master/apps/win/app.talon#L37

and we could stick with the global 'new window' command with VSCode
https://github.com/knausj85/knausj_talon/blob/606421ea22a321a22c44065de6133f396b81c1c4/misc/window_management.talon#L1

File: apps/win/vscode.talon [hunk 3]
```diff
@@ -6,6 +6,31 @@ app: Visual Studio Code
 app: Code.exe
 -
 
+# General
+action(user.ide_command_palette):
+  key(ctrl-shift-p)
+
+# Splits
+action(user.ide_split_right):
+  key(ctrl-shift-p)
```
Comment (knausj85) [inline apps/win/vscode.talon:15]: I'd suggest calling the user.ide_command_palette() here, so if the shortcut's just in one place.

File: apps/vscode.talon [hunk 4]
```diff
@@ -46,9 +46,9 @@ console up: user.ide_terminal_scroll_up()
 console down: user.ide_terminal_scroll_down()
```
Comment (knausj85) [inline apps/vscode.talon:46]: Do we still need any of these commands now that we're using ide.talon? Seems to be a lot of overlap, this should probably be reserved for anything that's specific to VSCode.

File: apps/win/vscode.talon (hunk 3 above)
Comment (auphofBSF) [reply apps/win/vscode.talon:15]: Agreed

File: code/ide.py (hunk 2 above)
Comment (auphofBSF) [reply code/ide.py:515]: Thanks that works

File: code/ide.py (hunk 1 above)
Comment (auphofBSF) [reply code/ide.py:278]: I felt there was enough Symantec difference to create this, but on checking a few IDE's I now have a different opinion.   
Yes in vscode this Explorer view maps to ide_toggle_project. However there is the ability to open an OS file explorer/manager and maybe this `ide_toggle_explorer`  should be that command `vscode> File:Reveal in file explorer` however it is not able to be toggled so it should finally be `ide_show_explorer`

additionally there are a number of other `ide_show_*` commands 
```
    def ide_show_explorer():
        """Show Explorer"""

    def ide_show_search():
        """Show Search"""

    def ide_show_source_control():
        """Show Source Control"""

    def ide_show_debug():
        """Show Debug"""

    def ide_show_extensions():
        """Show Extensions"""
```
I believe are hangovers from the merge of vscode.talon and ide.talon.   only the `ide_show_explorer` should stay for reason above.

I see you have done some great refactoring over weekend. I suggest I pull back this PR , cherry pick my updates on a new PR, I have also overweekend got my Linux Dev environment up and running ,so can test simultaneous against both platforms

Comment (knausj85) [reply code/ide.py:278]: Yeah, I agree with the ide_show_* stuff, good catch.

i can see the value in that ide_toggle_explorer/ide_show_explorer action, but would suggest something like `ide_reveal_in_file_manager`. Arguably a bit more descriptive/generic. 

Pulling this PR back sounds reasonable to me. Sorry for the inconvenience due to the refactor, been sitting on it for too long.

File: apps/vscode.talon (hunk 4 above)
Comment (auphofBSF) [reply apps/vscode.talon:46]: Agreed I think these are also legacy, I will attempt review  each and make changes via a  new PR as discussed

**Comment by auphofBSF** [comment]
As discussed, in code review, will be best with all vscode refactor introduced by bba1b56156eb264477ef73344751f91804f38957 to close this PR and resubmit on the new structure after with necessary changes discussed here

**Pull request closed without merging**